### PR TITLE
SDK Events

### PR DESCRIPTION
**Pull request by pmerlet-at-didomi**
Allow to listen to all native sdks events from Flutter

### CODE REVIEW

File: example/lib/events_helper.dart [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+
+    DidomiSdk.addEventListener(didomiListener);
+  }
+
+  void onEvent(eventDescription) {
+    uiCallback?.call(eventDescription);
+  }
+
+}
```
Comment (nicolas-chaix-didomi) [inline example/lib/events_helper.dart:108]: blank line is missing

File: lib/events/event_listener.dart [hunk 2]
```diff
@@ -9,4 +9,53 @@ class EventListener {
...
+  /// Purposes bulk action was switched to neutral
+  dynamic Function() onPreferencesClickResetAllPurposes = () {};
+  /// Vendors bulk action was switched to Agree
+  dynamic Function() onPreferencesClickAgreeToAllVendors = () {};
+  /// Vendors bulk action was switched to Disagree
+  dynamic Function() onPreferencesClickDisagreeToAllVendors = () {};
+  /// User consents synchronization was done
+  dynamic Function(String organizationUserId) onSyncDone = (organizationUserId) {};
+
 }
```
Comment (nicolas-chaix-didomi) [inline lib/events/event_listener.dart:61]: blank line is missing

File: lib/events/events_handler.dart [hunk 3]
```diff
@@ -19,13 +19,12 @@ class EventsHandler {
     try {
       handleDecodedEvent(json.decode(event));
     } on FormatException catch(e) {
-      print("Error while decoding event : $e");
+      print("Error while decoding event $event : $e");
```
Comment (nicolas-chaix-didomi) [inline lib/events/events_handler.dart:22]: there is no space before ':' in english
you could double them if you want
print("Error while decoding event $event: $e");
or
print("Error while decoding event $event :: $e");

Comment (pmerlet-at-didomi) [reply lib/events/events_handler.dart]: That's right, fixed

**Review by nicolas-chaix-didomi — APPROVED** [review]

**Pull request merged**